src/librustc_codegen_ssa/mir/statement.rs

```diff
@@ -17,7 +17,10 @@ impl<'a, 'tcx: 'a, Bx: BuilderMethods<'a, 'tcx>> FunctionCx<'a, 'tcx, Bx> {
         self.set_debug_loc(&mut bx, statement.source_info);
         match statement.kind {
             mir::StatementKind::Assign(ref place, ref rvalue) => {
-                if let mir::Place::Local(index) = *place {
+                if let mir::NeoPlace {
+                    base: mir::PlaceBase::Local(index),
+                    elems: &[],
+                }= *place {
```

Comment:
```rust
if let Some(local) = place.base_local() { ... }
```

Reply:
Yeah, this is code to be removed but yeah :).